libraries/crypto/src/ecdsa.rs

```diff
@@ -193,35 +193,40 @@ impl Signature {
         encoding
     }
 
-    #[cfg(feature = "std")]
-    pub fn from_bytes(bytes: &[u8]) -> Option<Signature> {
-        if bytes.len() != 64 {
-            None
-        } else {
-            let r =
-                NonZeroExponentP256::from_int_checked(Int256::from_bin(array_ref![bytes, 0, 32]));
-            let s =
-                NonZeroExponentP256::from_int_checked(Int256::from_bin(array_ref![bytes, 32, 32]));
-            if bool::from(r.is_none()) || bool::from(s.is_none()) {
-                return None;
-            }
-            let r = r.unwrap();
-            let s = s.unwrap();
-            Some(Signature { r, s })
+    pub fn from_bytes(bytes: &[u8; Signature::BYTES_LENGTH]) -> Option<Signature> {
+        let r_bytes_ref = array_ref![bytes, 0, int256::NBYTES];
+        let r = NonZeroExponentP256::from_int_checked(Int256::from_bin(r_bytes_ref));
+        let s_bytes_ref = array_ref![bytes, int256::NBYTES, int256::NBYTES];
+        let s = NonZeroExponentP256::from_int_checked(Int256::from_bin(s_bytes_ref));
+        if bool::from(r.is_none()) || bool::from(s.is_none()) {
+            return None;
         }
+        let r = r.unwrap();
+        let s = s.unwrap();
+        Some(Signature { r, s })
     }
 
-    #[cfg(test)]
-    fn to_bytes(&self, bytes: &mut [u8; 64]) {
-        self.r.to_int().to_bin(array_mut_ref![bytes, 0, 32]);
-        self.s.to_int().to_bin(array_mut_ref![bytes, 32, 32]);
+    #[cfg(feature = "std")]
+    pub fn to_bytes(&self, bytes: &mut [u8; Signature::BYTES_LENGTH]) {
+        self.r
+            .to_int()
+            .to_bin(array_mut_ref![bytes, 0, int256::NBYTES]);
+        self.s
+            .to_int()
+            .to_bin(array_mut_ref![bytes, int256::NBYTES, int256::NBYTES]);
     }
 }
 
 impl PubKey {
     #[cfg(feature = "with_ctap1")]
     const UNCOMPRESSED_LENGTH: usize = 1 + 2 * int256::NBYTES;
 
+    /// Creates a new PubKey from its coordinates on the elliptic curve.
+    pub fn from_coordinates(x: &[u8; NBYTES], y: &[u8; NBYTES]) -> Option<PubKey> {
+        PointP256::new_checked_vartime(Int256::from_bin(x), Int256::from_bin(y))
+            .map(|p| PubKey { p })
```

Comment:
IIRC you may be able to just write `.map(PubKey)`. But maybe this doesn't apply because `PubKey` is a struct rather than a tuple.

Reply:
The direct `map` doesn't work, but might if it had a suitable `PubKey::new`. I'd rather not implement one though.